build-tools/src/main/kotlin/org/jetbrains/kotlin/bitcode/CompileToBitcode.kt

```diff
@@ -70,7 +70,8 @@ open class CompileToBitcode @Inject constructor(
                 Language.C ->
                     // Used flags provided by original build of allocator C code.
                     listOf("-std=gnu11", "-O3", "-Wall", "-Wextra", "-Wno-unknown-pragmas",
```

Comment:
What about moving these flags to `runtime/build.gradle.kts`? There's already a `compilerArgs` override for `mimalloc`.
Here we can just leave
```kotlin
Language.C ->
    listOf("-std=gnu11", "-Werror", "-O3",
            "-Wall", "-Wextra")
```
to mimic C++ flags below (except for optimization level and language standard).